### ISSUE TITLE
Rule Change: Suggestions for prefer-regex-literals disallowRedundantWrapping option

### ISSUE THREAD

**Issue by yeonjuan** [eslint/eslint#16516]
### What rule do you want to change?

prefer-regex-literals

### What change to do you want to make?

Implement suggestions

### How do you think the change should be implemented?

Other

### Example code

```js
/* eslint prefer-regex-literals: ["error", {"disallowRedundantWrapping": true}] */
const reg0 = new RegExp('a'); 
const reg1 = new RegExp(/a/);   // no suggestion
const reg2 = new RegExp(/a/, 'g'); // no suggestion
```

### What does the rule currently do for this code?

The rule didn't suggest for unnecessarily wrapped patterns which disallowed by `"disallowRedundantWrapping": true`. [demo](https://eslint.org/play/#eyJ0ZXh0IjoiLyogZXNsaW50IHByZWZlci1yZWdleC1saXRlcmFsczogW1wiZXJyb3JcIiwge1wiZGlzYWxsb3dSZWR1bmRhbnRXcmFwcGluZ1wiOiB0cnVlfV0gKi9cblxuY29uc3QgYSA9IG5ldyBSZWdFeHAoL2EvLCAnZycpOyAgLy8gbm8gc3VnZ2VzdGlvblxuY29uc3QgYiA9IG5ldyBSZWdFeHAoXCJiXCIsICdnJyk7IC8vIGhhcyBhIHN1Z2dlc3Rpb24iLCJvcHRpb25zIjp7InBhcnNlck9wdGlvbnMiOnsiZWNtYVZlcnNpb24iOiJsYXRlc3QiLCJzb3VyY2VUeXBlIjoic2NyaXB0IiwiZWNtYUZlYXR1cmVzIjp7fX0sInJ1bGVzIjp7ImNvbnN0cnVjdG9yLXN1cGVyIjpbImVycm9yIl0sImZvci1kaXJlY3Rpb24iOlsiZXJyb3IiXSwiZ2V0dGVyLXJldHVybiI6WyJlcnJvciJdLCJuby1hc3luYy1wcm9taXNlLWV4ZWN1dG9yIjpbImVycm9yIl0sIm5vLWNhc2UtZGVjbGFyYXRpb25zIjpbImVycm9yIl0sIm5vLWNsYXNzLWFzc2lnbiI6WyJlcnJvciJdLCJuby1jb21wYXJlLW5lZy16ZXJvIjpbImVycm9yIl0sIm5vLWNvbmQtYXNzaWduIjpbImVycm9yIl0sIm5vLWNvbnN0LWFzc2lnbiI6WyJlcnJvciJdLCJuby1jb25zdGFudC1jb25kaXRpb24iOlsiZXJyb3IiXSwibm8tY29udHJvbC1yZWdleCI6WyJlcnJvciJdLCJuby1kZWJ1Z2dlciI6WyJlcnJvciJdLCJuby1kZWxldGUtdmFyIjpbImVycm9yIl0sIm5vLWR1cGUtYXJncyI6WyJlcnJvciJdLCJuby1kdXBlLWNsYXNzLW1lbWJlcnMiOlsiZXJyb3IiXSwibm8tZHVwZS1lbHNlLWlmIjpbImVycm9yIl0sIm5vLWR1cGUta2V5cyI6WyJlcnJvciJdLCJuby1kdXBsaWNhdGUtY2FzZSI6WyJlcnJvciJdLCJuby1lbXB0eSI6WyJlcnJvciJdLCJuby1lbXB0eS1jaGFyYWN0ZXItY2xhc3MiOlsiZXJyb3IiXSwibm8tZW1wdHktcGF0dGVybiI6WyJlcnJvciJdLCJuby1leC1hc3NpZ24iOlsiZXJyb3IiXSwibm8tZXh0cmEtYm9vbGVhbi1jYXN0IjpbImVycm9yIl0sIm5vLWV4dHJhLXNlbWkiOlsiZXJyb3IiXSwibm8tZmFsbHRocm91Z2giOlsiZXJyb3IiXSwibm8tZnVuYy1hc3NpZ24iOlsiZXJyb3IiXSwibm8tZ2xvYmFsLWFzc2lnbiI6WyJlcnJvciJdLCJuby1pbXBvcnQtYXNzaWduIjpbImVycm9yIl0sIm5vLWlubmVyLWRlY2xhcmF0aW9ucyI6WyJlcnJvciJdLCJuby1pbnZhbGlkLXJlZ2V4cCI6WyJlcnJvciJdLCJuby1pcnJlZ3VsYXItd2hpdGVzcGFjZSI6WyJlcnJvciJdLCJuby1sb3NzLW9mLXByZWNpc2lvbiI6WyJlcnJvciJdLCJuby1taXNsZWFkaW5nLWNoYXJhY3Rlci1jbGFzcyI6WyJlcnJvciJdLCJuby1taXhlZC1zcGFjZXMtYW5kLXRhYnMiOlsiZXJyb3IiXSwibm8tbmV3LXN5bWJvbCI6WyJlcnJvciJdLCJuby1ub25vY3RhbC1kZWNpbWFsLWVzY2FwZSI6WyJlcnJvciJdLCJuby1vYmotY2FsbHMiOlsiZXJyb3IiXSwibm8tb2N0YWwiOlsiZXJyb3IiXSwibm8tcHJvdG90eXBlLWJ1aWx0aW5zIjpbImVycm9yIl0sIm5vLXJlZGVjbGFyZSI6WyJlcnJvciJdLCJuby1yZWdleC1zcGFjZXMiOlsiZXJyb3IiXSwibm8tc2VsZi1hc3NpZ24iOlsiZXJyb3IiXSwibm8tc2V0dGVyLXJldHVybiI6WyJlcnJvciJdLCJuby1zaGFkb3ctcmVzdHJpY3RlZC1uYW1lcyI6WyJlcnJvciJdLCJuby1zcGFyc2UtYXJyYXlzIjpbImVycm9yIl0sIm5vLXRoaXMtYmVmb3JlLXN1cGVyIjpbImVycm9yIl0sIm5vLXVuZGVmIjpbImVycm9yIl0sIm5vLXVuZXhwZWN0ZWQtbXVsdGlsaW5lIjpbImVycm9yIl0sIm5vLXVucmVhY2hhYmxlIjpbImVycm9yIl0sIm5vLXVuc2FmZS1maW5hbGx5IjpbImVycm9yIl0sIm5vLXVuc2FmZS1uZWdhdGlvbiI6WyJlcnJvciJdLCJuby11bnNhZmUtb3B0aW9uYWwtY2hhaW5pbmciOlsiZXJyb3IiXSwibm8tdW51c2VkLWxhYmVscyI6WyJlcnJvciJdLCJuby11c2VsZXNzLWJhY2tyZWZlcmVuY2UiOlsiZXJyb3IiXSwibm8tdXNlbGVzcy1jYXRjaCI6WyJlcnJvciJdLCJuby11c2VsZXNzLWVzY2FwZSI6WyJlcnJvciJdLCJuby13aXRoIjpbImVycm9yIl0sInJlcXVpcmUteWllbGQiOlsiZXJyb3IiXSwidXNlLWlzbmFuIjpbImVycm9yIl0sInZhbGlkLXR5cGVvZiI6WyJlcnJvciJdfSwiZW52Ijp7ImVzNiI6dHJ1ZX19fQ==)

### What will the rule do after it's changed?

It will make suggestions for the patterns also.

```js
/* eslint prefer-regex-literals: ["error", {"disallowRedundantWrapping": true}] */
const reg1 = new RegExp(/a/);   //  =>  /a/
const reg2 = new RegExp(/a/, 'g'); // => /a/g
const reg3 = new RegExp(/a/g, 'i') // => /a/i
```

### Participation

- [X] I am willing to submit a pull request to implement this change.

**Comment by mdjermanovic** [comment]
This was originally included in https://github.com/eslint/eslint/issues/15029, but we decided to split the implementations into two PRs because https://github.com/eslint/eslint/pull/15077 was already complex enough, so marking accepted and PR is welcome!

**Comment by mdjermanovic** [comment]
> ```js
> const reg3 = new RegExp(/a/g, 'i') // => /a/i
> ```

I'm not sure about this one. While it's technically the same as `/a/i`, perhaps the intent was to have `/a/gi` so maybe we should offer two suggestions, or perhaps no suggestions for this case?

**Comment by yeonjuan** [comment]
IMO, `/a/i` seems better because it didn't change behavior of the code after accepting suggestions is not changed.

**Comment by mdjermanovic** [comment]
Yes, it would be the same behavior, but `new RegExp(/a/g, 'i')` looks like redundant wrapping + possible error. This is somewhat similar to the cases that `no-useless-escape` reports, where the rule always provides two suggestions. For example, for `"\a"` there's a suggestion to fix to `"a"` (the same behavior) and a suggestion to fix to `"\\a"` (not the same behavior, but it's very possible that this was the actual intent).

**Comment by mdjermanovic** [comment]
> ```js
> const reg1 = new RegExp(/a/);   //  =>  /a/
> const reg2 = new RegExp(/a/, 'g'); // => /a/g
> ```

As for these two cases, perhaps the rule could autofix them instead of providing suggestions?
